sulfhetnucrate.F90

```diff
@@ -17,24 +17,24 @@ subroutine sulfhetnucrate(carma, cstate, iz, igroup, nucbin, h2o, h2so4, beta1,
   use carmastate_mod
   use carma_mod
   use sulfate_utils
-  
+
   implicit none
-  
+
   type(carma_type), intent(in)         :: carma       !! the carma object
   type(carmastate_type), intent(inout) :: cstate      !! the carma state object
   integer, intent(in)                  :: iz          !! level index
   integer, intent(in)                  :: igroup      !! group index
   integer, intent(in)                  :: nucbin      !! bin in which nucleation occurs
-  real(kind=f), intent(in)             :: h2o         !! H2O concentrations in molec/cm3 
+  real(kind=f), intent(in)             :: h2o         !! H2O concentrations in molec/cm3
   real(kind=f), intent(in)             :: h2so4       !! H2SO4 concentrations in molec/cm3
   real(kind=f), intent(in)             :: beta1
   real(kind=f), intent(in)             :: beta2
   real(kind=f), intent(in)             :: ftry
-  real(kind=f), intent(in)             :: rstar       !! critical radius (cm)
+  real(kind=f), intent(inout)          :: rstar       !! critical radius (cm)
```

Comment:
Why is this inout? It isn't needed outside of this routine is it? Would you just need to make a local variable for rstar so you could calculate it when not supplied?

I was hoping Mike would pull out the part of Zhao that claculates ftry and rstar rather than putting a call to the whole routine here. THen you could probably do it all the time and not have ftry and rstar passed in here. Still seems kind of fragile if we have more routines coming in the future.